### DISCUSSION TITLE
Browser Controller Support

### DISCUSSION THREAD

**Discussion by szammyboi** [NicoHood/HID#358] [Q&A] [answered]
I have my Arduino Zero running as a gamepad on Windows. The joysticks and button work perfectly fine within Windows itself. However, in Chrome and Edge, the joysticks are showing up as only four values (-1.0, -0.9999, 0.9999, and 1.0). Is there a way to solve this programmatically or is it an HID driver issue?

**Comment by NicoHood** [comment, marked as answer]
Maybe it is related to this:
https://github.com/NicoHood/HID/issues/346

But nobody tried the suggested solution. Maybe you can.

**Reply by szammyboi** [reply]
I changed the range from (-32768, 32767) to (0, 65535) and it began working in the HTML5 browser API. It also still works in Windows so it completely solved the problem.

**Reply by szammyboi** [reply]
After further testing, it is a bit buggy. I can get it working on either the browser or Windows but not both at the same time.

**Reply by NicoHood** [reply]
This might be a problem of the driver. Maybe the driver is not updated properly. Try to connect to a different usb port, or checkout the wiki how to reinstall the driver.

Can you post a PR with your changes, so other people can review?

**Reply by szammyboi** [reply]
The behavior on Windows is odd. Just talking about the x-axis, when in the browser it gets mapped fine from -1 to 1. However, in the Windows joystick tester, the x-axis starts in the center but cannot go right. When you go left and hit 0, it then goes around to the other side and you can continue going until the center but no further. It seems like Windows just doesn't work with a fully positive range.
